addons/binding/org.openhab.binding.homepilot/src/main/java/org/openhab/binding/homepilot/discovery/HomePilotDeviceDiscoveryService.java

```diff
@@ -0,0 +1,162 @@
+/**
+ * Copyright (c) 2014-2016 by the respective copyright holders.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.homepilot.discovery;
+
+import static org.openhab.binding.homepilot.HomePilotBindingConstants.*;
+
+import java.util.List;
+import java.util.concurrent.CancellationException;
+import java.util.concurrent.Future;
+
+import org.eclipse.smarthome.config.discovery.AbstractDiscoveryService;
+import org.eclipse.smarthome.config.discovery.DiscoveryResult;
+import org.eclipse.smarthome.config.discovery.DiscoveryResultBuilder;
+import org.eclipse.smarthome.config.discovery.DiscoveryServiceCallback;
+import org.eclipse.smarthome.config.discovery.ExtendedDiscoveryService;
+import org.eclipse.smarthome.core.thing.ThingTypeUID;
+import org.eclipse.smarthome.core.thing.ThingUID;
+import org.openhab.binding.homepilot.handler.HomePilotBridgeHandler;
+import org.openhab.binding.homepilot.internal.HomePilotDevice;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.common.collect.ImmutableSet;
+
+/**
+ * The {@link HomePilotDeviceDiscoveryService} is used to discover devices that are connected to a Homematic gateway.
+ *
+ * @author Gerhard Riegler - Initial contribution
+ */
+public class HomePilotDeviceDiscoveryService extends AbstractDiscoveryService implements ExtendedDiscoveryService {
+    private static final Logger logger = LoggerFactory.getLogger(HomePilotDeviceDiscoveryService.class);
```

Comment:
Logger should NOT be static according to the guidelines. [[source](http://docs.openhab.org/developers/development/guidelines.html#e-logging)]